### PR TITLE
Widget deletion

### PR DESCRIPTION
**Pull request by djnunez-aot**
*Issue #143:*
https://github.com/bcgov/met-public/issues/143

-Added endpoints for front end

-Added deletion route for back end

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the met-public license (Apache 2.0).

### CODE REVIEW

**Review by saravanpa-aot — APPROVED** [review]

File: met-web/tests/unit/components/engagementForm.test.tsx [hunk 1]
```diff
@@ -299,7 +299,6 @@ describe('Engagement form page tests', () => {
         });
 
         expect(screen.getByText('Add Widget')).toBeVisible();
-        expect(screen.getByText('Who is Listening')).toBeVisible();
```
Comment (jadmsaadaot) [inline met-web/tests/unit/components/engagementForm.test.tsx:302]: I think if you changed to uppercase I for is this will work. "Who Is Listening"

Comment (djnunez-aot) [reply met-web/tests/unit/components/engagementForm.test.tsx:302]: ok ill try it now

File: met-api/src/met_api/models/widget.py [hunk 2]
```diff
@@ -74,3 +75,10 @@ def creat_all_widgets(cls, widgets: list) -> list[Widget]:
         db.session.add_all(new_widgets)
         db.session.commit()
         return new_widgets
+
+    @classmethod
+    def remove_widget(cls, engagement_id, widget_id,) -> DefaultMethodResult:
+        """Remove widget from engagement."""
+        widget = Widget.query.filter_by(id=widget_id, engagement_id=engagement_id).delete()
+        db.session.commit()
+        return DefaultMethodResult(True, 'Widget Removed', widget)
```
Comment (jadmsaadaot) [inline met-api/src/met_api/models/widget.py:84]: @saravanpa-aot prefers not to use DefaultMethodResult anymore. We could just return widget_id

File: met-api/src/met_api/services/widget_service.py [hunk 3]
```diff
@@ -92,3 +92,11 @@ def save_widget_items_bulk(self, widget_items: list, widget_id, user_id):
         self.create_added_widget_items(widget_items, widget_items_db, user_id)
         self.update_widget_items_sorting(widget_items, widget_id, user_id)
         return widget_items
+
+    @staticmethod
+    def delete_widget(engagement_id, widget_id):
+        """Remove widget from engagement."""
+        updated_widgets = Widget.remove_widget(engagement_id, widget_id)
```
Comment (jadmsaadaot) [inline met-api/src/met_api/services/widget_service.py:99]: updated_widgets?

File: met-api/src/met_api/models/widget.py (hunk 2 above)
Comment (saravanpa-aot) [reply met-api/src/met_api/models/widget.py:84]: just return widget itself

File: met-web/src/components/engagement/form/EngagementWidgets/WidgetCardSwitch.tsx [hunk 4]
```diff
@@ -1,26 +1,35 @@
...
-                    />
+                    <DragItem name="Who is Listening" moveItem={moveWidget} index={index}>
+                        <MetWidget
+                            testId={`who-is-listening-${widget.widget_type_id}`}
+                            title="Who is Listening"
+                            onDelete={() => {
+                                removeWidget(widget.id);
+                            }}
+                            onEdit={() => {
+                                handleWidgetDrawerOpen(true);
```
Comment (jadmsaadaot) [inline met-web/src/components/engagement/form/EngagementWidgets/WidgetCardSwitch.tsx:29]: consider opening the drawer and setting the tab to the who is listening form. Check if that works

**Review by jadmsaadaot — CHANGES_REQUESTED** [review]
Good job, looks clean. Minor comments

**Comment by jadmsaadaot** [comment]
Consider always merging main (which would reveal merge conflicts) to your local branch before creating a PR.

**Review by saravanpa-aot — APPROVED** [review]
Same comments as @jadmsaadaot for the backend stuff...I am good once you fix it

File: met-web/src/components/engagement/form/EngagementWidgets/WidgetCardSwitch.tsx (hunk 4 above)
Comment (djnunez-aot) [reply met-web/src/components/engagement/form/EngagementWidgets/WidgetCardSwitch.tsx:31]: great idea it worked

File: met-api/src/met_api/services/widget_service.py [hunk 5]
```diff
@@ -92,3 +92,11 @@ def save_widget_items_bulk(self, widget_items: list, widget_id, user_id):
...
+        if not updated_widgets:
+            raise ValueError('Widget to update was not found')
```
[the remark this answers is not in the document]
Comment (djnunez-aot) [reply met-api/src/met_api/services/widget_service.py]: fixed

File: met-api/src/met_api/services/widget_service.py (hunk 3 above)
Comment (djnunez-aot) [reply met-api/src/met_api/services/widget_service.py]: i was doing updating and removing simultaneously so just cleaned that up

File: met-api/src/met_api/models/widget.py (hunk 2 above)
Comment (djnunez-aot) [reply met-api/src/met_api/models/widget.py]: perfect

File: met-web/tests/unit/components/engagementForm.test.tsx (hunk 1 above)
Comment (djnunez-aot) [reply met-web/tests/unit/components/engagementForm.test.tsx]: seems it did not work not sure why

**Review by jadmsaadaot — APPROVED** [review]

**Pull request merged**